tests/test_connection_pool.py

```diff
@@ -122,25 +129,28 @@ def test_connection_pool_blocks_until_timeout(self):
         # we should have waited at least 0.1 seconds
         assert time.time() - start >= 0.1
 
-    def connection_pool_blocks_until_another_connection_released(self):
+    def test_connection_pool_blocks_until_another_connection_released(self, master_host):  # noqa: E501
         """
         When out of connections, block until another connection is released
         to the pool
         """
-        pool = self.get_pool(max_connections=1, timeout=2)
+        connection_kwargs = {'host': master_host}
+        pool = self.get_pool(max_connections=1, timeout=2,
+                             connection_kwargs=connection_kwargs)
         c1 = pool.get_connection('_')
 
         def target():
             time.sleep(0.1)
             pool.release(c1)
 
-        Thread(target=target).start()
         start = time.time()
+        Thread(target=target).start()
```

Comment:
This test is failing repeatedly in all interpreters, from within a container and without, because the timing is sometimes wrong. (It's inconsistent.) I think it's because we start the thread before starting the timer? This version of the test hasn't failed yet.